### PR TITLE
Refactoring the monitor and env threads to use PeriodicOperations

### PR DESCRIPTION
**Pull request by narrieta**

### CODE REVIEW

File: azurelinuxagent/ga/collect_logs.py [hunk 1]
```diff
@@ -85,10 +84,7 @@ def __init__(self):
         self.event_thread = None
         self.should_run = True
         self.last_state = None
-
```
Comment (narrieta) [inline azurelinuxagent/ga/collect_logs.py:88]: There was no need to use PeriodicOperations in the log collector thread. This was probably copy and pasted from the Monitor thread.

File: azurelinuxagent/ga/collect_telemetry_events.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ class ExtensionEventSchema(object):
     OperationId = "OperationId"
 
 
-class _ProcessExtensionEventsPeriodicOperation(PeriodicOperation):
```
Comment (narrieta) [inline azurelinuxagent/ga/collect_telemetry_events.py:71]: I removed the "PeriodicOperation" and "Operation" suffixes across all the usages of PeriodicOperation to improve readability

File: azurelinuxagent/ga/periodic_operation.py [hunk 3]
```diff
@@ -33,9 +33,8 @@ class PeriodicOperation(object):
     # To prevent flooding the log with error messages we report failures at most every hour
     _LOG_WARNING_PERIOD = datetime.timedelta(minutes=60)
 
-    def __init__(self, name, operation, period):
-        self._name = name
-        self._operation = operation
+    def __init__(self, period):
```
Comment (narrieta) [inline azurelinuxagent/ga/periodic_operation.py:36]: Now that all the code is refactored to use subclasses of PeriodicOperation, there is no need of a string to identify the operation (we can use the type) nor the need to pass the function to execute as parameter (this is now done using inheritance)

**Review by larohra — APPROVED** [review]
Thanks for the refactor!

File: azurelinuxagent/ga/collect_logs.py (hunk 1 above)
Comment (narrieta) [reply azurelinuxagent/ga/collect_logs.py:88]: The thread blocks waiting for the log collector. If we did this on one of the other threads we may end up blocking all the operations in that thread while the log collector runs. With a little more of extra code we could just start the collector and not block on it, similarly to what we do when we start extensions. I think the code was written like this to make it a little simpler. We can always improve on it later.

**Pull request merged**